war/src/main/js/components/tooltips/index.js

```diff
@@ -0,0 +1,119 @@
+import tippy from "tippy.js";
+import behaviorShim from "@/util/behavior-shim";
+
+const TOOLTIP_BASE = {
+  arrow: false,
+  theme: "tooltip",
+  animation: "tooltip",
+  appendTo: document.body,
+};
+
+let tooltipInstances = [];
+const globalPlugin = {
+  fn(instance) {
+    return {
+      onCreate() {
+        tooltipInstances = tooltipInstances.concat(instance);
+      },
+      onDestroy() {
+        tooltipInstances = tooltipInstances.filter((i) => i !== instance);
+      },
+    };
+  },
+};
+
+tippy.setDefaultProps({
+  plugins: [globalPlugin],
+});
+
+/**
+ * Registers tooltips for the page
+ * If called again, destroys existing tooltips and registers them again (useful for progressive rendering)
+ * @param {HTMLElement} container - Registers the tooltips for the given container
+ */
+function registerTooltips(container) {
+  if (!container) {
+    container = document;
+  }
+
+  tooltipInstances.forEach((instance) => {
+    if (instance.props.container === container) {
+      instance.destroy();
+    }
+  });
+
+  tippy(
+    container.querySelectorAll(
+      '[tooltip]:not([tooltip=""]):not([data-html-tooltip])'
+    ),
+    Object.assign(
+      {
+        content: (element) =>
+          element.getAttribute("tooltip").replace(/<br[ /]?\/?>|\\n/g, "\n"),
+        container: container,
+        onCreate(instance) {
+          instance.reference.setAttribute("title", instance.props.content);
+        },
+        onShow(instance) {
+          instance.reference.removeAttribute("title");
+        },
+        onHidden(instance) {
+          instance.reference.setAttribute("title", instance.props.content);
+        },
+      },
+      TOOLTIP_BASE
+    )
+  );
+
+  tippy(
+    container.querySelectorAll("[data-html-tooltip]"),
+    Object.assign(
+      {
+        content: (element) => element.getAttribute("data-html-tooltip"),
```

Comment:
matrix auth will need adapting to this for ATH to pass I think.

I created a draft awhile back but it's conflicted currently: https://github.com/jenkinsci/matrix-auth-plugin/pull/125